Userland/Utilities/nc.cpp

```diff
@@ -144,16 +146,24 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
         TRY(Core::System::setsockopt(fd, SOL_SOCKET, SO_RCVTIMEO, &timeout, sizeof(timeout)));
         TRY(Core::System::setsockopt(fd, SOL_SOCKET, SO_SNDTIMEO, &timeout, sizeof(timeout)));
 
-        auto* hostent = gethostbyname(target.characters());
-        if (!hostent) {
-            warnln("Socket::connect: Unable to resolve '{}'", target);
-            return 1;
-        }
-
         sockaddr_in dst_addr {};
         dst_addr.sin_family = AF_INET;
         dst_addr.sin_port = htons(port);
-        dst_addr.sin_addr.s_addr = *(in_addr_t const*)hostent->h_addr_list[0];
+
+        if (!numeric_mode) {
+            auto* hostent = gethostbyname(target.characters());
+            if (!hostent) {
+                warnln("Socket::connect: Unable to resolve '{}'", target);
```

Comment:
This is not something that you introduced, but `Socket::connect` is no longer correct here (I assume the original code was borrowed from there).